bin/publish.js

```diff
@@ -347,28 +370,47 @@ async function main() {
     // --force-publish ensures that all packages release a new version regardless
     // of whether changes have occurred in them
     // --yes skips the prompt for confirming the version
-    nextVersion = retrieveNextVersion();
-    execWithLog(`lerna version ${nextVersion} --force-publish --exact --yes`, true);
+    nextVersion = retrieveNextVersion(options);
+    let lernaCommand = `lerna version ${nextVersion} --force-publish --exact --yes`;
+    if (options.dryRun) {
+      lernaCommand += ' --no-git-tag-version --no-push';
+    }
+
+    execWithLog(lernaCommand, true);
     console.log(`✅ ` + chalk.cyan(`Successfully Versioned ${nextVersion}`));
   } else {
     console.log('⚠️ ' + chalk.grey(`Skipping Versioning`));
   }
+
   if (!options.skipPack) {
     cleanProject();
     packAllPackages();
     console.log(`✅ ` + chalk.cyan(`Successfully Packaged ${nextVersion}`));
   } else {
     console.log('⚠️ ' + chalk.grey(`Skipping Packaging`));
   }
+
   if (!options.skipPublish) {
     const tarballs = collectTarballPaths();
-    await confirmPublish(tarballs, nextVersion);
+    const npmAuthTokenInEnv = !!process.env.NODE_AUTH_TOKEN;
+    if (!npmAuthTokenInEnv && !options.dryRun) {
+      if (process.env.CI) {
+        throw new Error('No NODE_AUTH_TOKEN environment variable, cannot continue publishing.');
+      }
+    }
+    // Assume human ran script if token is missing
+    await confirmPublish(tarballs, options, !npmAuthTokenInEnv);
     console.log(`✅ ` + chalk.cyan(`Successfully Published ${nextVersion}`));
   } else {
     console.log('⚠️ ' + chalk.grey(`Skipping Publishing`));
   }
 }
 
+let cli = readline.createInterface({
```

Comment:
is this left over from something? I don't see it used in this PR

Reply:
good catch, reverted